api/base/filters.py

```diff
@@ -292,13 +292,28 @@ def parse_query_params(self, query_params):
                             field_name: self._parse_date_param(field, source_field_name, op, value),
                         })
                     elif not isinstance(value, int) and source_field_name in ['_id', 'guid._id', 'journal_id', 'moderation_state']:
-                        query.get(key).update({
-                            field_name: {
-                                'op': 'in',
-                                'value': self.bulk_get_values(value, field),
-                                'source_field_name': source_field_name,
-                            },
-                        })
+                        if VersionedGuidMixin.GUID_VERSION_DELIMITER in value:
+                            base_guid, version = value.split(VersionedGuidMixin.GUID_VERSION_DELIMITER)
+                            query.get(key).update({
+                                'id': {
+                                    'op': 'eq',
+                                    'value': base_guid,
+                                    'source_field_name': 'versioned_guids___id',
+                                },
+                                'versioned_guids__version': {
+                                    'op': 'eq',
+                                    'value': version,
+                                    'source_field_name': 'versioned_guids__version',
+                                },
+                            })
```

Comment:
Needs to handle different cases:

* Error if `.split_guid()` returns `None, None`
* Don't query version if returns `base_guid, None`
* Query both guid and version if returns `base_guid, version`